src/core/gui/SearchBar.cpp

```diff
@@ -23,33 +23,37 @@ SearchBar::SearchBar(Control* control): control(control) {
 
     GtkWidget* next = win->get("btSearchForward");
     GtkWidget* previous = win->get("btSearchBack");
-    g_signal_connect(next, "clicked", G_CALLBACK(+[](GtkButton* button, SearchBar* self) { self->searchNext(); }),
-                     this);
+    g_signal_connect(next, "clicked",
+                     G_CALLBACK(+[](GtkButton*, gpointer d) { static_cast<SearchBar*>(d)->searchNext(); }), this);
```

Comment:
Use our safe casts here, this is actually no plain callback. The same applies for the next ~20 lines.